### ISSUE TITLE
TrivialAccessors should ignore class instance methods.

### ISSUE THREAD

**Issue by dnl** [bbatsov/rubocop#2281]
```
class Thing
  def self.thing=(value)
    @thing = value
  end
end
```
was flagged by TrivialAccessors, it should ignore def self.whatever as `attr_accessor` is only for instance methods. (and this is not a rails project so has no `[cm]attr_accessor`)

(there's probably a very good reason not to do this in the first place ;)

`rubocop -V => 0.34.2 (using Parser 2.2.2.6, running on ruby 2.2.2 x86_64-darwin14)`

**Comment by alexdowad** [comment]
`TrivialAccessors` was deliberately designed to flag `def self.thing=(value)` as a "trivial accessor". You can turn this off using the `IgnoreClassMethods` config parameter for this cop.

You don't need Rails to use `attr_accessor` on a class:

```ruby
class Thing
  class << self
    attr_accessor :thing
  end
end
```

...does the job just fine.

**Comment by alexdowad** [comment]
This should be closed.
